### PR TITLE
feat: 판매자 승인 신청 기능 구현

### LINKED ISSUE #3: feat(seller): 판매자 승인 신청 API 추가
#### 📝 어떤 기능을 추가하나요?
회원이 판매자 등록 요청을 보낼 수 있도록 합니다.

#### 👀 자세한 내용
- 사업자 등록번호 제출
- 기본 상태는 PENDING
- 관리자 승인 후 상품 등록 가능

### PR DESCRIPTION
**Pull request by perArdua**
Closes #3

### 📌 작업 개요
판매자 승인 신청 기능을 DDD 방식으로 구현했습니다.  
회원이 사업자 등록번호를 제출하면 `Seller`로 등록되며, 승인 상태는 기본적으로 `PENDING`입니다.  
도메인 생성 책임과 정책은 내부 팩토리 메서드에 위임하고, 예외 및 테스트를 분리하여 구성했습니다.

---

### ✅ 주요 변경 사항

- `user.controller.seller`  
  - `SellerController`: `/api/v1/sellers/apply` API 처리

- `user.service.seller`  
  - `SellerCommandService`, `SellerCommandServiceImpl`: 판매자 등록 로직 처리

- `user.dto.request.seller`  
  - `SellerApplyRequest`: 판매자 신청 요청 DTO

- `user.domain`  
  - `Seller`: 도메인 정의 및 팩토리 메서드 구성  
  - `SellerRepository`: 판매자 중복 여부 조회 및 저장소 인터페이스 정의

- `user.exception.seller`  
  - `SellerAlreadyAppliedException`: 중복 신청 예외  
  - `SellerErrorCode`: 판매자 예외 코드 정의

---

### 🧪 테스트

- `SellerCommandServiceImplTest`: 판매자 신청 성공/중복 예외 테스트
- `SellerControllerTest`: API 요청/응답 테스트 포함
- `SellerTest`: Seller 생성 및 상태 확인 도메인 테스트
- `SellerFixture`: 테스트용 기본 Seller 객체 제공

---

### 📎 관련 이슈
- #3

### CODE REVIEW

**Comment by perArdua** [comment]
@codecov-ai-reviewer review

**Comment by ghost** [comment]
## PR Description

This pull request introduces the functionality for users to apply to become sellers within the Hopping Mall application. It allows authenticated users to submit a seller application, which includes their business registration number. The system validates the application and persists the seller information.

<details>
<summary><b>Click to see more</b></summary>

### Key Technical Changes
1.  **Seller Entity:** Created a `Seller` entity to represent seller information, including a one-to-one relationship with the `User` entity and an `ApprovalStatus` enum.
2.  **Seller Application API:** Implemented a `/sellers/apply` endpoint in `SellerController` that accepts a `SellerApplyRequest` containing the business registration number.
3.  **Business Number Validation:** Added validation to prevent duplicate business registration numbers and ensure the provided number is not blank.
4.  **Seller Command Service:** Implemented `SellerCommandService` and `SellerCommandServiceImpl` to handle the business logic for seller application, including checking for existing applications and saving the new seller information.
5.  **Exception Handling:** Introduced custom exceptions (`SellerAlreadyAppliedException`, `SellerBusinessNumberDuplicateException`, `SellerNotFoundException`) and corresponding error codes to handle specific seller-related errors.
6.  **Repository:** Created `SellerRepository` to handle database interactions for the `Seller` entity.

### Architecture Decisions
1.  **One-to-One Relationship:** A one-to-one relationship between `User` and `Seller` was chosen to represent that each user can have at most one seller profile.
2.  **Approval Status Enum:** An `ApprovalStatus` enum was used to manage the different states of a seller application (PENDING, APPROVED, REJECTED).
3.  **Command Service Pattern:** A command service pattern was implemented to encapsulate the seller application logic, promoting separation of concerns and testability.

### Dependencies and Interactions
1.  **User Entity:** The `Seller` entity depends on the `User` entity to associate the seller profile with a specific user.
2.  **UserRepository:** The `SellerCommandServiceImpl` interacts with the `UserRepository` to retrieve the user applying to become a seller.
3.  **Security Context:** The `SellerController` relies on Spring Security's `@AuthenticationPrincipal` to retrieve the currently authenticated user's details.
4.  **ApiResponse:** The `SellerController` uses the `ApiResponse` class for consistent response formatting.

### Risk Considerations
1.  **Business Number Uniqueness:** The system relies on the database to enforce the uniqueness of business registration numbers. Ensure proper indexing and constraints are in place.
2.  **User ID Conversion:** The `SellerController` converts the username to a Long for the user ID. This assumes the username is always a valid Long and could lead to errors if the username format changes. Input validation is needed.
3.  **Security:** Ensure that only authenticated users with appropriate roles (e.g., SELLER) can access the `/sellers/apply` endpoint. Proper authorization mechanisms should be in place.

### Notable Implementation Details
1.  The `SellerCommandServiceImpl` checks for existing seller applications and duplicate business registration numbers before creating a new seller entity.
2.  Custom exceptions are thrown to provide specific error messages to the client.
3.  Comprehensive unit tests are included to verify the functionality of the controller, service, and domain logic.
</details>

File: src/main/kotlin/com/hoppingmall/mall/user/service/seller/SellerCommandServiceImpl.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        if (sellerRepository.findNullableByUserId(userId) != null) {
+            throw SellerAlreadyAppliedException()
+        }
+
+        if (sellerRepository.existsByBusinessNumber(request.businessNumber)) {
+            throw SellerBusinessNumberDuplicateException()
+        }
+
+        val seller = Seller.create(user, request.businessNumber)
```
Comment (ghost) [inline src/main/kotlin/com/hoppingmall/mall/user/service/seller/SellerCommandServiceImpl.kt:32]: Consider checking for business number duplication before checking if user already applied. This provides better user experience by giving more specific error messages first.
```suggestion
    @Transactional
    override fun apply(userId: Long, request: SellerApplyRequest) {
        val user = userRepository.findNullableById(userId)
            ?: throw UserNotFoundException()

        if (sellerRepository.existsByBusinessNumber(request.businessNumber)) {
            throw SellerBusinessNumberDuplicateException()
        }

        if (sellerRepository.findNullableByUserId(userId) != null) {
            throw SellerAlreadyAppliedException()
        }

        val seller = Seller.create(user, request.businessNumber)
        sellerRepository.save(seller)
    }
```

File: src/test/kotlin/com/hoppingmall/mall/user/domain/SellerTest.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            password = Password("encoded"),
+            name = "판매자",
+            role = Role.SELLER
+        )
+
+        val seller = Seller.fixture(user = user)
+
+        assertEquals(Seller.ApprovalStatus.PENDING, seller.getApprovalStatus())
+        assertEquals("123-45-67890", seller.businessNumber)
+        assertEquals(user, seller.user)
```
Comment (ghost) [inline src/test/kotlin/com/hoppingmall/mall/user/domain/SellerTest.kt:25]: Consider testing edge cases such as business number validation and testing the immutability of certain fields after creation.
```suggestion
    @Test
    fun `Seller 생성 시 기본 승인 상태는 PENDING이다`() {
        val user = User.fixture(
            email = Email("seller@example.com"),
            password = Password("encoded"),
            name = "판매자",
            role = Role.SELLER
        )

        val seller = Seller.fixture(user = user)

        assertEquals(Seller.ApprovalStatus.PENDING, seller.getApprovalStatus())
        assertEquals("123-45-67890", seller.businessNumber)
        assertEquals(user, seller.user)
        // Test immutability
        assertSame(user, seller.user)
    }
```

File: src/test/kotlin/com/hoppingmall/mall/user/service/seller/SellerCommandServiceImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        whenever(sellerRepository.save(captor.capture())).thenReturn(mock())
+
+        // when
+        sellerCommandService.apply(userId, request)
+
+        // then
+        verify(sellerRepository).save(any())
+        val savedSeller = captor.firstValue
+        assert(savedSeller.user == user)
+        assert(savedSeller.businessNumber == request.businessNumber)
```
Comment (ghost) [inline src/test/kotlin/com/hoppingmall/mall/user/service/seller/SellerCommandServiceImplTest.kt:41]: The test method name could be more descriptive about what specific aspect is being tested. Also consider testing the order of validations.
```suggestion
    @Test
    fun `유효한 사용자가 중복되지 않은 사업자번호로 판매자 신청하면 Seller 엔티티가 저장된다`() {
```
